File: backend/modules/commands/command_parser.go
package commands

import (
	"jarvis-ai/modules/agent"
	"log"
	"strings"
)

// normalizeInput lowercases the input and strips the wake-word prefix
// in all three supported languages (English, Hindi, Marathi).
func normalizeInput(original string) string {
	input := strings.ToLower(strings.TrimSpace(original))

	// Strip wake-word prefix — all language variants
	wakeWords := []string{
		"jarvis ", "hey jarvis ", "ok jarvis ", "अरे जार्विस ", "जार्विस ", "hey rudra ", "rudra ",
	}
	for _, ww := range wakeWords {
		input = strings.TrimPrefix(input, ww)
	}
	return strings.TrimSpace(input)
}

func ParseAndExecute(input string) string {
	normalized := normalizeInput(input)
	log.Printf("[CommandRouter] Normalized request: %s", normalized)

	// Dynamically delegate request execution directly to the specialized Agent Orchestrator
	response, err := agent.ExecuteAgent(normalized)
	if err != nil {
		log.Println("[CommandRouter] Error invoking Agent Orchestrator:", err)
		return "I encountered an error executing that request. Please verify that Ollama is running locally."
	}
	
	return response
}

File: backend/app/services/command_parser.py
import logging
from app.services.agent_orchestrator import execute_agent

logger = logging.getLogger(__name__)

def normalize_input(original: str) -> str:
    original = original.lower().strip()
    wake_words = [
        "jarvis ", "hey jarvis ", "ok jarvis ", "अरे जार्विस ", "जार्विस ", "hey rudra ", "rudra "
    ]
    for ww in wake_words:
        if original.startswith(ww):
            original = original[len(ww):]
    return original.strip()

async def parse_and_execute(input_text: str) -> str:
    normalized = normalize_input(input_text)
    logger.info(f"[CommandRouter] Normalized request: {normalized}")
    
    try:
        response = await execute_agent(normalized)
        return response
    except Exception as e:
        logger.error(f"[CommandRouter] Error invoking Agent Orchestrator: {e}")
        return "I encountered an error executing that request. Please verify that Ollama is running locally."
